doc/fpmsyncd/hld_fpmsyncd-NTT.md

```diff
@@ -0,0 +1,571 @@
+  <tr><td>RTA_NH_ID</td><td>NextHop Group ID</td></tr>
+</table>
+
+#### Example of entries in APPL_DB
+
+For example. following route configuration will generate events show in the table below:
+
+```
+S>* 8.8.8.0/24 [1/0] via 10.0.1.5, Ethernet4, weight 1, 00:01:09 
+  * via 10.0.2.6, Ethernet8, weight 1, 00:01:09 
+S>* 9.9.9.0/24 [1/0] via 10.0.1.5, Ethernet4, weight 1, 00:00:04 
+  * via 10.0.2.6, Ethernet8, weight 1, 00:00:04
+```
+<table>
+  <tr><td>Seq</td><td>Event</td><td>Attributes</td><td>Value</td></tr>
+  <tr><td rowspan="3">1</td><td rowspan="3">RTM_NEWNEXTHOP</td><td>NHA_ID</td><td>116</td></tr>
+  <tr><td>NHA_GATEWAY</td><td>10.0.1.5</td></tr>
+  <tr><td>NHA_OIF</td><td>22</td></tr>
+  <tr><td rowspan="3">2</td><td rowspan="3">RTM_NEWNEXTHOP</td><td>NHA_ID</td><td>117</td></tr>
+  <tr><td>NHA_GATEWAY</td><td>10.0.2.6</td></tr>
+  <tr><td>NHA_OIF</td><td>23</td></tr>
+  <tr><td rowspan="2">3</td><td rowspan="2">RTM_NEWNEXTHOP</td><td>NHA_ID</td><td>118</td></tr>
+  <tr><td>NHA_GROUP</td><td>[{116,1},{117,1}]</td></tr>
+  <tr><td rowspan="2">4</td><td rowspan="2">RTM_NEWROUTE</td><td>RTA_DST</td><td>8.8.8.0/24</td></tr>
+  <tr><td>RTA_NH_ID</td><td>118</td></tr>
+  <tr><td rowspan="2">5</td><td rowspan="2">RTM_NEWROUTE</td><td>RTA_DST</td><td>9.9.9.0/24</td></tr>
+  <tr><td>RTA_NH_ID</td><td>118</td></tr>
+</table>
+
+A short description of `fpmsyncd` logic flow:
+
+- When receiving `RTM_NEWNEXTHOP` events on sequence 1, 2 and 3, `fpmsyncd` will save the information in an internal list to be used when necessary.
+- When `fpmsyncd` receive the `RTM_NEWROUTE` on sequence 4, the process will write the NextHop Group with ID 118 to the `NEXTHOP_GROUP_TABLE` using the information of gateway and interface from the NextHop Group events with IDs 116 and 117.
+- Then `fpmsyncd` will create a new route entry to `ROUTE_TABLE` with a `nexthop_group` field with value `ID118`.
+- When `fpmsyncd` receives the last `RTM_NEWROUTE` on sequence 5, the process will create a new route entry (but no NextHop Group entry) in `ROUTE_TABLE` with `nexthop_group` field with value `ID118`. (Note: This NextHop Group entry was created when the `fpmsyncd` received the event sequence 4.)
+
+#### Example of entries in ASIC_DB
+
+The `ASIC_DB` entry is not changed by this enhancement.
+Therefore, even after this enhancement, table entries will be created for `ROUTE_ENTRY`, `NEXT_HOP_GROUP`, `NEXT_HOP_GROUP_MEMBER`, and `NEXT_HOP` respectively, as shown in the example below
+
+<!-- omit in toc -->
+##### Figure: Example of ASIC_DB entry
+![fig3](diagram/fig3.svg)
+
+
+### SAI API 
+
+No changes are being made in SAI.
+The end result of what gets programmed via SAI will be the same as current implementation when manually adding `NEXTHOP_GROUP_TABLE` entries to `APPL_DB`.
+
+
+### Configuration and management 
+
+This NextHop Group feature is enabled/disabled by config option of zebra (BGP container): `[no] fpm use-next-hop-groups`
```

Comment:
can you follow example here?

https://github.com/sonic-net/SONiC/blob/master/doc/BGP/BGP-supress-fib-pending.md#632--device_metadata

Reply:
Yes. @nakano-omw is working on moving enabled/disabled config under DEVICE_METADATA.

Reply:
Updated both HLD and Source Code to config use DEVICE_METADATA.